### PR TITLE
Refactor tracing into logging mod

### PR DESCRIPTION
**Pull request by jkitman**
Refactors tracing into logging mod for `fedimint-server`.

Simplifies some of our dependencies and logging setup code.

### CODE REVIEW

File: fedimint-server/src/logging.rs [hunk 1]
```diff
@@ -1,6 +1,95 @@
...
+
+        let chrome_layer_opt = || -> Option<Box<dyn Layer<_> + Send + Sync + 'static>> {
+            #[cfg(feature = "telemetry")]
+            if self.with_chrome {
+                let (cr_layer, gaurd) = tracing_chrome::ChromeLayerBuilder::new()
+                    .include_args(true)
+                    .build();
+                // drop gaurd cause file to written and closed
+                // in this case file will closed after exit of program
+                std::mem::forget(gaurd);
```
Comment (justinmoon) [inline fedimint-server/src/logging.rs:80]: *guard?

File: integrationtests/tests/fixtures/mod.rs [hunk 2]
```diff
@@ -141,23 +140,8 @@ pub async fn fixtures(num_peers: u16) -> anyhow::Result<Fixtures> {
...
-            tracing_subscriber::registry().with(cr_layer).init();
-        } else {
-            tracing_subscriber::fmt()
-                .with_env_filter(
-                    EnvFilter::try_from_default_env()
-                        .unwrap_or_else(|_| EnvFilter::new("info,fedimint::consensus=warn")),
-                )
-                .init();
-        }
+        let chrome = env::var_os("FEDIMINT_TRACE_CHROME").map_or(false, |x| !x.is_empty());
```
Comment (justinmoon) [inline integrationtests/tests/fixtures/mod.rs:143]: Kinda tangential, but should we standardize our environment variables on `FM_` prefix?

**Review by justinmoon — APPROVED** [review]

**Review by justinmoon — APPROVED** [review]

File: fedimint-server/src/logging.rs (hunk 1 above)
Comment (Maan2003) [reply fedimint-server/src/logging.rs:80]: this is fine and tracing chrome is only used in integration testing.
see https://github.com/fedimint/fedimint/pull/1372#discussion_r1072863036

**Review by Maan2003 — APPROVED** [review]
LGTM

**Review by dpc — APPROVED** [review]

**Comment by elsirion** [comment]
This should probably become its own crate, e.g. the gateway should use the same code.

**Pull request merged**